### ISSUE TITLE
Network Security Improvement Proposal

### ISSUE THREAD

**Issue by c29r3** [solana-labs/solana#21801]
#### Problem
In Solana, it is quite easy to match a validator with its ip address and find out when a particular validator will be the leader. Knowing this data, you can carry out a selective attack on large validators (DDoS).

#### Proposed Solution
Perhaps it makes sense to complicate the mapping between validator_address-ip_address?
For example, in Tendermint, the queue of validators is also known in advance and who will produce the block, but it is quite difficult to find out the ip address of the validator.

**Comment by askibin** [comment]
it looks like Tendermint uses private network to connect validators. Solana validators are not in single cloud infrastructure, it would not be easy to put all nodes behind VPN. Also, it would undermine decentralization.

**Comment by c29r3** [comment]
Tendermint does not use VPN. Validators and full nodes are connected by specifying the addresses of seed nodes and persistent peers. 
The point here is different. That the Cosmos SDK does not have a method with which you can match the validator address with its ip address

**Comment by askibin** [comment]
I didn't say Tendermint uses VPN, I said they use private connections. From here https://docs.tendermint.com/master/nodes/validators.html:
`All sentry nodes (the “frontend”) connect to the validator using this private connection. The validator does not have a public IP address to provide its services.
Amazon has multiple availability zones within a region. One can install sentry nodes in other regions too. In this case the second, third and further regions need to have a private connection to the validator node. This can be achieved by VPC Peering (“VPC Network Peering” in Google Cloud).`
Don't get me wrong, I'm with you on this topic, the question is - how you can hide ips of validators connected thru a public network using open source software and no information held privately by a 3rd party? You can obfuscate a little, but one can always come up with a tool that lists all ips anyway.

**Comment by c29r3** [comment]
Ah, okay, now I understand you.
Sentry is optional. When I gave Tendermint as an example, I did not consider this option, it is not used very often.
There are reasons for this:
- infrastructure is more expensive
- the support of the validator / updates becomes more complicated
- the process of identifying errors becomes more complicated
- the validator is actually connected through its sentry nodes, and if there are problems with all of them, then it goes offline

It is not necessary to hide the validator's address or the IP address, but to complicate the way to find the validator's ip address using validator_address \ vote_key
To do this, it is enough to hide the ip address or pubkey in the method:
```bash
curl http://localhost:8899 -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0", "id":1, "method":"getClusterNodes"}' | jq .result[1]
```
OUTPUT
```
{
  "featureSet": 52865992,
  "gossip": "88.198.41.60:8001",
  "pubkey": "98vNehxpVCPz1t5RjnZZVjxQ549a2VLUbkpNwqj1rSb8",
  "rpc": null,
  "shredVersion": 8573,
  "tpu": "88.198.41.60:8004",
  "version": "1.8.10"
}
```

But this will break all existing monitoring systems of validators that consider the concentration of stakes in data centers...
